### PR TITLE
Update to match beta release

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
**Incomplete, do not merge!**

@garanj perhaps you could have a light look over this and see if there are any larger/architectural things that could be improved? There's some obvious things that need to be changed before the actual release, like the path to the beta location.

FYI @sarahamilton @breanatate 

To build from AOSP:

1. Download source code [as described here](https://android.googlesource.com/platform/frameworks/support/+/androidx-main/docs/onboarding.md#source).
1. Change into the `frameworks/support` directory.
1. Run `./gradlew createArchive`.
1. This should create `../../out/androidx/build/support_repo` (i.e. in the `out`
subdirectory of `androidx-main`).
1. To create zip of maven artifacts:
a. `cd out/androidx/build/support_repo`
a. `zip -r ~/m2repo androidx/health/health-services-client`
1. Unzip this file into e.g. `/tmp`:
a. `cd /tmp`
a. `unzip ~/m2repo`
1. Add this directory to the `maven { url … }` stanza in `build.gradle`, e.g.:
```txt
allprojects {
  repositories {
    maven { url("/tmp/m2repo") }
    google()
    mavenCentral()
  }
}
```

(To resync to head, run `repo sync`.)

### CODE REVIEW

File: health-services/ExerciseSample/app/src/main/java/com/example/exercise/HealthServicesManager.kt [hunk 1]
```diff
@@ -200,26 +188,33 @@ class HealthServicesManager @Inject constructor(
...
             }
 
-            override fun onLapSummary(lapSummary: ExerciseLapSummary) {
+            override fun onLapSummaryReceived(lapSummary: ExerciseLapSummary) {
                 coroutineScope.runCatching {
                     trySendBlocking(ExerciseMessage.LapSummaryMessage(lapSummary))
                 }
             }
 
+            override fun onRegistered() {
```
Comment (garanj) [inline health-services/ExerciseSample/app/src/main/java/com/example/exercise/HealthServicesManager.kt:211]: What do we recommend apps do in `onRegistered` and `onRegistrationFailed`? Is there anything sensible apps can do?

File: health-services/PassiveData/app/src/main/java/com/example/passivedata/PassiveDataRepository.kt [hunk 2]
```diff
@@ -42,11 +48,12 @@ class PassiveDataRepository @Inject constructor(
         }
     }
 
-    val lastestHeartRate: Flow<Double> = dataStore.data.map { prefs ->
+    val latestHeartRate: Flow<Double> = dataStore.data.map { prefs ->
         prefs[LATEST_HEART_RATE] ?: 0.0
     }
 
     suspend fun storeLatestHeartRate(heartRate: Double) {
+        Log.d("qqqqqq", "storing heartRate $heartRate")
```
Comment (garanj) [inline health-services/PassiveData/app/src/main/java/com/example/passivedata/PassiveDataRepository.kt:56]: Probably can rename the tag here? :)

File: health-services/PassiveData/app/src/main/java/com/example/passivedata/PassiveDataService.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    override fun onNewDataPointsReceived(dataPoints: List<DataPoint>) {
+        runBlocking {
+            dataPoints.latestHeartRate()?.let {
+                repository.storeLatestHeartRate(it)
+            }
+        }
+    }
+
+    override fun onRegistered() {
+        TODO("Not yet implemented")
```
Comment (garanj) [inline health-services/PassiveData/app/src/main/java/com/example/passivedata/PassiveDataService.kt:25]: Anything useful that  apps can do in `onRegistered` / `onRegistrationFailed`?

File: health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/HealthServicesManager.kt [hunk 4]
```diff
@@ -64,26 +70,44 @@ class HealthServicesManager @Inject constructor(
...
     suspend fun subscribeForGoals() {
         Log.i(TAG, "Subscribing for goals")
-        val componentName = ComponentName(context, PassiveGoalsReceiver::class.java)
-        // Each goal is a separate subscription.
-        passiveMonitoringClient.registerPassiveGoalCallback(dailyStepsGoal, componentName).await()
-        passiveMonitoringClient.registerPassiveGoalCallback(floorsGoal, componentName).await()
+        val passiveListenerConfig = PassiveListenerConfig.builder()
+            .setPassiveGoals(setOf(dailyStepsGoal, floorsGoal, hrGoal))
+            .build()
+        val future = passiveMonitoringClient.setPassiveListenerServiceAsync(
```
Comment (garanj) [inline health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/HealthServicesManager.kt:90]: Can avoid futures using await() in a try / catch?

File: health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/PassiveGoalService.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
+package com.example.passivegoals
+
+import android.util.Log
+import androidx.health.services.client.PassiveListenerService
+import androidx.health.services.client.data.*
```
Comment (garanj) [inline health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/PassiveGoalService.kt:5]: Nit: Avoid wildcard imports (don't know the full reasons why, do you?). Can set AS up to do this,

File: health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/PassiveGoalService.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                    repository.setLatestDailyGoalAchieved(time)
+                }
+        }
+    }
+
+    override fun onPermissionLost() {
+        Log.d("qqqqq", "permission lost")
+    }
+
+    override fun onRegistered() {
```
Comment (garanj) [inline health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/PassiveGoalService.kt:55]: So looks like these are part of the Service so need to be overridden, but no need to override `onNewDataPointsReceived` in this example etc as they are from the `PassiveListenerCallback` interface and just limit it to `onGoalCompleted`?

File: health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/PassiveGoalService.kt (hunk 5 above)
Comment (ithinkihaveacat) [reply health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/PassiveGoalService.kt:5]: Fixed.

File: health-services/PassiveData/app/src/main/java/com/example/passivedata/PassiveDataRepository.kt (hunk 2 above)
Comment (ithinkihaveacat) [reply health-services/PassiveData/app/src/main/java/com/example/passivedata/PassiveDataRepository.kt:56]: Oops, removed!

**Comment by ithinkihaveacat** [comment]
Split into separate pull requests for each sample and superseded by #54 (and other pull requests to follow).

**Pull request closed without merging**